client/landing/stepper/declarative-flow/internals/hooks/use-preload-steps.ts

```diff
@@ -0,0 +1,77 @@
+import { SiteDetails } from '@automattic/data-stores';
+import debugFactory from 'debug';
+import { useEffect } from 'react';
+import { useSelector } from 'calypso/state';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import type { Flow, StepperStep } from '../types';
+
+const debug = debugFactory( 'calypso:stepper:preloading' );
+
+async function tryPreload( step?: StepperStep, followingStep?: StepperStep ) {
+	if ( step && 'asyncComponent' in step ) {
+		debug( 'Preloading step:', step.slug );
+		await step.asyncComponent();
+	}
+	// Flows are indeterminate, they often pick one of the two next steps based on user input, so load two steps ahead.
+	tryPreload( followingStep );
```

Comment:
we have to check whether the `followingStep` exists or not. Otherwise, it calls infinitely

Reply:
No, because the function starts with an `if` that does the check.